kore/src/Kore/Step/SMT/Translate.hs

```diff
@@ -82,6 +92,9 @@ data TranslateItem variable
     = QuantifiedVariable !(ElementVariable variable)
     | UninterpretedTerm !(TermLike variable)
 
+type TranslateTerm variable m =
+    SExpr -> TranslateItem variable -> Translator variable m SExpr
```

Comment:
Is a `TranslateTerm` function actually allowed to depend on the type of `m`?

Reply:
I'm not entirely sure I understand the question right, but [translateTerm](https://github.com/kframework/kore/blob/deba447b057f856f5141e8ba18d42561d0c456eb/kore/src/Kore/Step/SMT/Evaluator.hs#L222) is the function we eventually pass to `translatePredicateWith`, and it's polymorphic in `m`.